### PR TITLE
bugfix/ZBUG-4523

### PR DESCRIPTION
**Pull request by ashwinsahu19**
Added "from" address in the session object. As validation is failing the code flow for scheduled mail.

### CODE REVIEW

File: store/src/java/com/zimbra/cs/service/mail/SaveDraft.java [hunk 1]
```diff
@@ -170,10 +171,15 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
 
             if (autoSendTime != 0) {
                 AccountUtil.checkQuotaWhenSendMail(mbox);
+                Session mSession=null;
                 try {
+                    mSession=JMSession.getSmtpSession(mbox.getAccount());
+                    if (mailAddress != null) {
+                        mSession.getProperties().setProperty("mail.smtp.from", mailAddress);
```
Comment (mamtadahiwal) [inline store/src/java/com/zimbra/cs/service/mail/SaveDraft.java]: mail.smtp.from declare this as  constant string --> private static final String MAIL_SMTP_FROM = "mail.smtp.from";

Comment (ashwinsahu19) [reply store/src/java/com/zimbra/cs/service/mail/SaveDraft.java]: Created a constant for the property.

**Review by mamtadahiwal — APPROVED** [review]

File: store/src/java/com/zimbra/cs/service/mail/SaveDraft.java [hunk 2]
```diff
@@ -170,10 +172,15 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
 
             if (autoSendTime != 0) {
                 AccountUtil.checkQuotaWhenSendMail(mbox);
+                Session mSession=null;
                 try {
+                    mSession=JMSession.getSmtpSession(mbox.getAccount());
+                    if (mailAddress != null) {
```
Comment (chitranshu198) [inline store/src/java/com/zimbra/cs/service/mail/SaveDraft.java]:
```suggestion
                    if (null != mailAddress) {
```

File: store/src/java/com/zimbra/cs/service/mail/SaveDraft.java [hunk 3]
```diff
@@ -170,10 +172,15 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
 
             if (autoSendTime != 0) {
                 AccountUtil.checkQuotaWhenSendMail(mbox);
+                Session mSession=null;
```
Comment (chitranshu198) [inline store/src/java/com/zimbra/cs/service/mail/SaveDraft.java]:
```suggestion
                Session mSession = null;
```

File: store/src/java/com/zimbra/cs/service/mail/SaveDraft.java (hunk 2 above)
Comment (ashwinsahu19) [reply store/src/java/com/zimbra/cs/service/mail/SaveDraft.java]: done

File: store/src/java/com/zimbra/cs/service/mail/SaveDraft.java (hunk 3 above)
Comment (ashwinsahu19) [reply store/src/java/com/zimbra/cs/service/mail/SaveDraft.java]: done

**Review by chitranshu198 — APPROVED** [review]

File: store/src/java/com/zimbra/cs/service/mail/SaveDraft.java [hunk 4]
```diff
@@ -170,10 +172,15 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
 
             if (autoSendTime != 0) {
                 AccountUtil.checkQuotaWhenSendMail(mbox);
+                Session mSession = null;
                 try {
+                    mSession=JMSession.getSmtpSession(mbox.getAccount());
```
Comment (dasiyogesh) [inline store/src/java/com/zimbra/cs/service/mail/SaveDraft.java]: add space before & after `=`

**Review by dasiyogesh — APPROVED** [review]

File: store/src/java/com/zimbra/cs/service/mail/SaveDraft.java (hunk 4 above)
Comment (ashwinsahu19) [reply store/src/java/com/zimbra/cs/service/mail/SaveDraft.java]: Made required changes.

**Pull request merged**